pkg/executor/kubernetes.go

```diff
@@ -457,7 +473,12 @@ func (th *k8sTaskHandle) ExitCode() (int, error) {
 // false if the pod didn't terminate before the provided timeout.
 func (th *k8sTaskHandle) Wait(timeout time.Duration) (bool, error) {
 	if th.isTerminated() {
-		return true, nil
+		select {
```

Comment:
This situation won't happen when fully initialized TaskHandle is returned from Execute.
Could we take this and paste it in Execute? Then we can think how to simplify it :)